test/unit/range/container/constexpr_vector_test.cpp

```diff
@@ -0,0 +1,365 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/range/container/constexpr_vector.hpp>
+#include <seqan3/range/container/constexpr_string.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/std/ranges>
+
+#include <utility>
+#include <string>
+
+using namespace std::literals;
+using namespace seqan3;
+
+// standard construction.
+TEST(constexpr_vector, standard_construction)
+{
+    EXPECT_TRUE((std::is_default_constructible_v<constexpr_vector<char, 4>>));
+    // EXPECT_TRUE((std::is_trivially_default_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_nothrow_default_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_copy_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_trivially_copy_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_nothrow_copy_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_move_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_trivially_move_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_nothrow_move_constructible_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_copy_assignable_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_trivially_copy_assignable_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_nothrow_copy_assignable_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_move_assignable_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_trivially_move_assignable_v<constexpr_vector<char, 4>>));
+    EXPECT_TRUE((std::is_nothrow_move_assignable_v<constexpr_vector<char, 4>>));
+}
+
+TEST(constexpr_vector, concepts)
+{
+    EXPECT_TRUE((container_concept<constexpr_vector<char, 4>>));
```

Comment:
what's missing for `reservable_container_concept`? (stuff like `reserve` exists, although a no-op)